### PR TITLE
Society and Industrial stage button bars fixes and editor button into its own scene

### PR DESCRIPTION
**Pull request by CI09**
**Brief Description of What This PR Does**

Also fixes editor button anchor points

**Related Issues**

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [ ] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/gui_common/EditorEntryButton.tscn
Comment (hhyyrylainen) [inline src/gui_common/EditorEntryButton.tscn:1]: This file should not be in gui common, this should be in the microbe stage gui folder. To follow the rule that a specific GUI part used for a stage should be in the stage's folder where it first appears. I don't consider this a general GUI thing as not all stages will use this.

File: src/gui_common/EditorEntryButton.tscn [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+tracks/1/interp = 1
+tracks/1/loop_wrap = true
+tracks/1/keys = {
+"times": PackedFloat32Array(0.5, 2.1, 4.4, 5),
+"transitions": PackedFloat32Array(1, 2, 1, 2),
+"update": 0,
+"values": [0.0, 0.8, 0.8, 0.0]
+}
+
+[sub_resource type="AnimationLibrary" id="AnimationLibrary_xveqw"]
```
Comment (hhyyrylainen) [inline src/gui_common/EditorEntryButton.tscn:40]: As this is now a separate scene, it is pretty questionable how `CreatureStageHUDBase.OnEditorButtonMouseEnter` works. So I think that script functionality should be split into a separate file and class called EditoryEntryButton.cs.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/gui_common/EditorEntryButton.tscn (hunk 1 above)
Comment (hhyyrylainen) [reply src/gui_common/EditorEntryButton.tscn:40]: That does stuff like this:
```
        editorButton.GetNode<TextureRect>("Highlight").Hide();
        editorButton.GetNode<AnimationPlayer>("AnimationPlayer").Stop();
```

So it explicitly refers to the child nodes of a separate scene. That's a big problem in terms of encapsulation, reuse and decoupling. This needs to be redone with a EditorEntryButton class that handles modifying its internal child nodes.

Comment (hhyyrylainen) [reply src/gui_common/EditorEntryButton.tscn:40]: What do you mean? I'm saying you should remove the mouse enter and exit signals from the separate HUD scenes and put them in the new editor entry scene, then write a C# class that implements that functionality for that scene.

Comment (hhyyrylainen) [reply src/gui_common/EditorEntryButton.tscn:40]: That was the only immediate thing I saw. If the root node of the new scene is a button, then letting the button click registration and enable / disable work like they currently do is perfectly fine to me.

File: src/microbe_stage/gui/EditorEntryButton.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    [Export]
+    private AnimationPlayer buttonAnimationPlayer = null!;
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            if (highlight != null)
+            {
+                highlight.Dispose();
```
Comment (hhyyrylainen) [inline src/microbe_stage/gui/EditorEntryButton.cs:20]: These should not be disposed, these aren't NodePaths.

Comment (CI09) [reply src/microbe_stage/gui/EditorEntryButton.cs:20]: My IDE told me these should be.

Comment (hhyyrylainen) [reply src/microbe_stage/gui/EditorEntryButton.cs:20]: Follow our styleguide and add a pragma warning suppression around the two variables.

**Review by hhyyrylainen — APPROVED** [review]
Code change looks fine to me now.

**Comment by hhyyrylainen** [comment]
I'll just merge this without testing as the big double check of everything in the game has not been done yet.

**Pull request merged**